aptos-move/aptos-gas-calibration/src/gas_meter_helpers.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::gas_meter::GasMeters;
+use aptos::move_tool::MemberId;
+use aptos_cached_packages::aptos_stdlib;
+use aptos_framework::BuiltPackage;
+use aptos_language_e2e_tests::{account::Account, executor::FakeExecutor};
+use aptos_types::transaction::TransactionPayload;
+use move_binary_format::CompiledModule;
+use move_core_types::{account_address::AccountAddress, language_storage::ModuleId};
+use std::{fs::ReadDir, path::PathBuf, string::String, time::Instant};
+
+//// CONSTANTS
+const PREFIX: &str = "calibrate_";
+
+/// Generate a TransactionPayload for modules
+///
+/// ### Arguments
+///
+/// * `package` - Built Move package
+pub fn generate_module_payload(package: &BuiltPackage) -> TransactionPayload {
+    // extract package data
+    let code = package.extract_code();
+    let metadata = package
+        .extract_metadata()
+        .expect("extracting package metadata must succeed");
+
+    // publish package similar to create_publish_package in harness.rs
+    aptos_stdlib::code_publish_package_txn(
+        bcs::to_bytes(&metadata).expect("PackageMetadata has BCS"),
+        code,
+    )
+}
+
+/// Sign transaction to create a Module and return transaction status
+///
+/// ### Arguments
+///
+/// * `executor` - Runs transactions
+/// * `account` - Account to publish module under
+/// * `payload` - Info relating to the module
+/// * `sequence_number` - Nonce
+pub fn sign_module_txn(
```

Comment:
The name is confusing. When we say "sign a transaction", we only mean generating and attaching a cryptographic signature to the transaction. This function however, also executes the transaction.

Reply:
Renamed to `execute_user_txn`

Reply:
Err I renamed `sign_user_txn` instead of `sign_module_txn`. Both should now be renamed.